github-actions/github-release/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Node dependencies
+ */
+const fs = require( 'fs' );
+
+/**
+ * GitHub dependencies
+ */
+const core = require( '@actions/core' );
+const github = require( '@actions/github' );
+
+( async () => {
+	try {
+		const repoToken = core.getInput( 'repo-token', { required: true } );
+		const zipPath = core.getInput( 'zip', { required: true } );
+
+		const client = new github.GitHub( repoToken );
+
+		const gitRef = github.context.ref;
+		let gitTag = gitRef.match( /refs\/tags\/(.*)/ );
+
+		if ( null === gitTag ) {
+			throw new Error( `Failed to determine tag from git ref: ${ gitRef }` );
+		}
+
+		gitTag = gitTag[ 1 ];
+
+		core.info( `Fetching release details for git tag: ${ gitTag }` );
+
+		const { owner, repo } = github.context.payload;
+
+		const releaseInfo = {};
+
+		try {
+			// An error will be thrown if a release for the tag was not found.
+			const currentReleaseInfo = await client.repos.getReleaseByTag( { owner, repo, tag: gitTag } );
+
+			core.info( 'Tag was previously released, deleting it' );
+
+			releaseInfo.tag_name = currentReleaseInfo.data.tag_name;
+			releaseInfo.name = currentReleaseInfo.data.name;
+			releaseInfo.body = currentReleaseInfo.data.body;
+			await client.repos.deleteRelease( { owner, repo, release_id: currentReleaseInfo.data.id } );
+		} catch ( error ) {
+			core.info( 'Release not found for tag' );
+
+			releaseInfo.tag_name = gitTag;
+			releaseInfo.name = gitTag;
```

Comment:
The name here should be the version that is contained inside of `amp.php` in `amp.zip`. For example: `1.5.0-alpha-20191121T050254Z-be678184`. This would be important for plugins like the beta tester to be checking for updates. If the name of the release doesn't match the current version, then it would know it needs to update.